src/components/GiveProject/GrantCard.tsx

```diff
@@ -84,41 +78,9 @@ export default function GrantCard({ grant, mode }: GrantDetailsProps) {
   const dispatch = useAppDispatch();
 
   useEffect(() => {
-    // We use dispatch to asynchronously fetch the results, and then update state variables so that the component refreshes
-    // We DO NOT use dispatch here, because it will overwrite the state variables in the redux store, which then creates havoc
-    // e.g. the redeem yield page will show someone else's deposited sOHM and redeemable yield
-    getRedemptionBalancesAsync({
-      networkID: networkId,
-      provider: provider,
-      address: wallet,
-    })
-      .then(resultAction => {
-        setTotalDebt(resultAction.redeeming.recipientInfo.totalDebt);
-        setRecipientInfoIsLoading(false);
-      })
-      .catch(e => console.log(e));
-
-    getDonorNumbers({
-      networkID: networkId,
-      provider: provider,
-      address: wallet,
-    })
-      .then(resultAction => {
-        setDonorCount(!resultAction ? 0 : resultAction.length);
-        setDonorCountIsLoading(false);
-      })
-      .catch(e => console.log(e));
-
-    getTotalDonated({
-      networkID: networkId,
-      provider: provider,
-      address: wallet,
-    })
-      .then(donatedAmount => {
-        setTotalDonated(donatedAmount);
-      })
-      .catch(e => console.log(e));
-  }, [connected, networkId, isGiveModalOpen]);
+    setIsUserDonating(false);
+    setDonationId(NO_DONATION);
+  }, [networkId]);
```

Comment:
If the networkId changes, don't we also need to invalidate the fetched data? Unless you've included the networkId as part of the react-query key.

Reply:
(You have, so ignore this.)